template-xml/data/src/main/java/co/nimblehq/template/xml/data/repository/UserPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,54 @@
+package co.nimblehq.template.xml.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.template.xml.domain.model.UserPreferences
+import co.nimblehq.template.xml.domain.repository.UserPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class UserPreferencesRepositoryImpl @Inject constructor(private val dataStore: DataStore<Preferences>) :
+    UserPreferencesRepository {
+
+    private val TAG: String = "UserPreferencesRepo"
```

Comment:
[Optional] 
Normally when defining `TAG` I usually put it in the `companion object` block and I see that in `PreferencesRepositoryImpl` you declared it the same way, so I think it may be better if we keep it consistently 🤔 
```suggestion
       companion object {
        private const val TAG = "UserPreferencesRepo"
    }
```

Reply:
Ditto here with the usage of `canonicalName`/`simpleName`
```
Log.e(UserPreferencesRepositoryImpl::class.java.canonicalName, "Error reading preferences.", exception)
```